### PR TITLE
DOC: special: update docstrings to reflect C translation of AMOS

### PR DESCRIPTION
**Pull request by j-bowhay**
I think we should probably update docstring to reflect the changes that have happened under the hood. Feel free to rephrase as you see fit

### CODE REVIEW

File: scipy/special/_amos.c [hunk 1]
```diff
@@ -977,7 +977,6 @@ double complex amos_airy(
         *nz = amos_bknu(zta, fnu, kode, 1, &cy[0], tol, elim, alim);
     }
     s1 = cy[0]*coef;
-
```
Comment (j-bowhay) [inline scipy/special/_amos.c]: Oops not sure how I picked these up, will revert

**Comment by ilayn** [comment]
Oh indeed, I completely forgot all this :smile:, thanks.

It seems to me that we don't need to bother folks with the translation detail or in fact all that z letters which is the type of the argument, a typical Fortranism for float, double and complex variants `s`, `d`, `c`, `z`. What do you think ? 

By the way, it seems like you are not on the latest state of the branch, I still see Cython etc. commits

**Comment by j-bowhay** [comment]
> Oh indeed, I completely forgot all this 😄, thanks.

No worries I think it might also be worth updating the docstring for functions that use cfdlib too. Happy to open a PR on main.

> It seems to me that we don't need to bother folks with the translation detail or in fact all that z letters which is the type of the argument, a typical Fortranism for float, double and complex variants `s`, `d`, `c`, `z`. What do you think ?

Happy to slim down the description but think it is worth keeping the type letter as `cairy` and `zairy` have slightly different implementations, for example, so it might be useful for the sake of one letter to have all the information. How about ``Wrapper for a translation of the AMOS [1]_ routines `zairy` and `zbiry`.`` for example?

> By the way, it seems like you are not on the latest state of the branch, I still see Cython etc. commits

Yes realised I should have rebased rather than merged.

**Comment by ilayn** [comment]
> How about `Wrapper for a translation of the AMOS [1]_ routines` `zairy` and `zbiry`. for example?

I'm a bit leaning towards `uses ``zairy``, ``zbiry`` from AMOS [1]_`. Because probably nobody cares what or how we used the functions. But don't have any strong opinions really. I'd surprised if people even know what AMOS is anymore :) So pick which one sounds right for you. 

>  I think it might also be worth updating the docstring for functions that use cfdlib too. Happy to open a PR on main

Indeed.

**Comment by ilayn** [comment]
Albert pushed the button a bit too early before we can react 😂 So i think you can freely open proper PRs now

**Pull request closed without merging**